app/services/mapping/goodsshipment/AEOMutualRecognitionPartiesBuilder.scala

```diff
@@ -16,42 +16,24 @@
 
 package services.mapping.goodsshipment
 
-import java.util
-
 import forms.declaration.DeclarationAdditionalActors
 import javax.inject.Inject
-import models.declaration.DeclarationAdditionalActorsData
 import services.mapping.ModifyingBuilder
-import uk.gov.hmrc.http.cache.client.CacheMap
+import services.mapping.goodsshipment.AEOMutualRecognitionPartiesBuilder.{createAdditionalActors, isDefined}
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 import wco.datamodel.wco.declaration_ds.dms._2._
 
-import scala.collection.JavaConverters._
-
 class AEOMutualRecognitionPartiesBuilder @Inject()()
     extends ModifyingBuilder[DeclarationAdditionalActors, GoodsShipment] {
 
   override def buildThenAdd(model: DeclarationAdditionalActors, goodsShipment: GoodsShipment): Unit =
-    if (AEOMutualRecognitionPartiesBuilder.isDefined(model))
-      goodsShipment.getAEOMutualRecognitionParty.add(AEOMutualRecognitionPartiesBuilder.createAdditionalActors(model))
+    if (isDefined(model))
+      goodsShipment.getAEOMutualRecognitionParty.add(createAdditionalActors(model))
 
 }
 
 object AEOMutualRecognitionPartiesBuilder {
 
-  def build(implicit cacheMap: CacheMap): util.List[GoodsShipment.AEOMutualRecognitionParty] =
-    cacheMap
-      .getEntry[DeclarationAdditionalActorsData](DeclarationAdditionalActorsData.formId)
-      .filter(_.actors.nonEmpty)
-      .map(
-        _.actors
-          .filter(actor => isDefined(actor))
-          .map(createAdditionalActors)
-      )
-      .getOrElse(Seq.empty)
-      .toList
-      .asJava
-
   private def isDefined(actor: DeclarationAdditionalActors): Boolean =
```

Comment:
We can move across these `isDefined` methods to the class now, they were only in the object because the old `build` method needed access to them statically

Reply:
👍